dojo_plugin/pages/canvas.py

```diff
@@ -0,0 +1,225 @@
+                                        students[grade_res["user_id"]], grade_credit_percent)
+                
+                return res
+            # multi-module and mutli-user sync of grades with canvas using bulk update api
+            else:
+                # additional check to be sure that they align 
+                if grade_res['assessment_grades'][aindex]['module_id'] == assessment['id']:
+                    student_id = students[grade_res['user_id']]
+                    canvas_grade_data[f"sis_user_id:{student_id}"] = {"posted_grade": grade_credit_percent}
+                    assessment_student_counter[assessment['id']] += 1
+                else:
+                    log.error(f"The following did not align in list, {grade_res['assessment_grades'][aindex]['module_id']=}   {assessment['id']=}" )
+                    return {"status": "failed", "message": f"The following did not align in list, {grade_res['assessment_grades'][aindex]['module_id']=}   {assessment['id']=}, this shouldn't happen."}
+
+        if (len(canvas_grade_data) > 0):
+            submission_results = post_bulk_grade_data_to_canvas(canvas_token, canvas_api_host, canvas_course_id, canvas_assignment_id, canvas_grade_data)
+            if submission_results["status"] == "success":
+                progress_urls.append(submission_results["url"])
+            else:
+                posting_results.append(submission_results)            
+        
+    if len(progress_urls) > 0 :
+        check_results = check_progress(progress_urls, canvas_token, interval=2)
+        posting_results.extend(check_results)
+        posting_results.append(assessment_student_counter)
+
+    return posting_results
+
+
+def post_grade_to_canvas(canvas_token, canvas_api_host, canvas_course_id, assignment_id, student_id, students_grade):
+    url = f"https://{canvas_api_host}/api/v1/courses/{canvas_course_id}/assignments/{assignment_id}/submissions/sis_user_id:{student_id}"
+    payload = {'submission': {'posted_grade': students_grade}}
+    
+    header = {'Authorization': 'Bearer ' + canvas_token}
+    
+    r = requests.put(url, headers=header, json=payload)
+    
+    if r.status_code == 200:
+        return {"status": "success", "student_id": student_id, "json": r.json()}
+    else:
+        return {"student_id": student_id, "code": r.status_code, "text": r.text, "url": url, "canvas_token": canvas_token}
+    
+
+def post_bulk_grade_data_to_canvas(canvas_token, canvas_api_host, canvas_course_id, assignment_id, canvas_grade_data):
+    url = f"https://{canvas_api_host}/api/v1/courses/{canvas_course_id}/assignments/{assignment_id}/submissions/update_grades"
+    payload = {'grade_data': canvas_grade_data}
+    
+    header = {
+                'Authorization': 'Bearer ' + canvas_token,
+                "Content-Type": "application/json"
+            }
+    
+    r = requests.post(url, headers=header, json=payload)
+    
+    r_json = r.json()
+    
+    if r.status_code == 200:
```

Comment:
The logic in the following doesn't mirror `post_grade_to_canvas`, is there a reason for this? If not, let's have them both match.